### ISSUE TITLE
Missing Screen Reader announcement for multi-select combo box

### ISSUE THREAD

**Issue by jake-costa** [jpmorganchase/salt-ds#5723]
### Latest version

- [x] I have tested the latest version

### Description

For the multi-select combo box, after making multiple selections using Jaws or NVDA and pressing backspace, the first item is announced as removed/unselected. However, any subsequent removals using backspace are not conveyed to the user.

This is working correctly for voiceover

### Steps to reproduce

1. Turn on Jaws/NVDA
2. Open either Chrome/Firefox
3. Navigate to [Salt Multi-Select Combobox](https://www.saltdesignsystem.com/salt/components/combo-box/examples#multi-select)
4. Select 2+ options
5. After selecting the last option press backspace (Jaws/NVDA will announce unselected)
6. Press backspace again to remove the other option(s)
7. The removal of those options are not announced.

### Expected behavior

A removal announcement should be made for every option that is removed.

Works correctly with Mac/Safari/VoiceOver

### Package name(s)

Core (@salt-ds/core)

### Browser

- [x] Chrome
- [ ] Safari
- [x] Firefox
- [ ] Microsoft Edge

### Operating system

- [ ] macOS
- [x] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Are you a JPMorgan Chase & Co. employee?

- [x] I am an employee of JPMorgan Chase & Co.

**Comment by bhoppers2008** [comment]
Migrated to Jira.
